### ISSUE TITLE
calling any? with block over error.details is leaving empty keys

### ISSUE THREAD

**Issue by h0jeZvgoxFepBQ2C** [rails/rails#38528]
I found a slight strange behaviour by using the error details and querying some specific keys for our system..

Same happens with plain ActiveRecord-Models.

### Steps to reproduce

```ruby
class User
  include ActiveModel::Model

  attr_accessor :email
  validates :email, presence: true
end

a = User.new;a.valid?;a.errors.details[:email].any?{|a| a[:email] == :blank}
a.errors.details
=> {:email=>[{:error=>:blank}]}

a = User.new;a.valid?;a.errors.details[:firstname].any?{|b| b.last == :blank}
a.errors.details
=> {:email=>[{:error=>:blank}], :firstname=>[]}
# why is here an firstname key? It shouldn't be there, since I only called any? ?
```

### Expected behavior
The error details should have only an error key.

### Actual behavior
The error details have an empty firstname key, even though it should not be there.

### System configuration
**Rails version**: 5.2

**Ruby version**: 2.3.4

**Comment by h0jeZvgoxFepBQ2C** [comment]
Actually it gets even more strange...
Just by looking up the keys, it's creating the keys..

```ruby
[28] pry(main)> a.errors[:test123]
=> []

[29] pry(main)> a.errors
=> #<ActiveModel::Errors:0x007f9cbbe03260
 @base=#<User:0x007f9cbbe03378 @errors=#<ActiveModel::Errors:0x007f9cbbe03260 ...>, @validation_context=nil>,
 @details={:email=>[{:error=>:blank}], :firstname=>[]},
 @messages={:email=>["can't be blank"], :test123=>[]}>
# why is here now a test123 key?

[30] pry(main)> a.errors.details[:test345345]
=> []
# why is here now a test345345 key?


[31] pry(main)> a.errors
=> #<ActiveModel::Errors:0x007f9cbbe03260
 @base=#<User:0x007f9cbbe03378 @errors=#<ActiveModel::Errors:0x007f9cbbe03260 ...>, @validation_context=nil>,
 @details={:email=>[{:error=>:blank}], :firstname=>[], :test345345=>[]},
 @messages={:email=>["can't be blank"], :test123=>[]}>
```

**Comment by joshmn** [comment]
You might want to consider using `include?(attribute)` instead of `any?` — https://github.com/rails/rails/blob/master/activemodel/lib/active_model/errors.rb#L177

Per your comment about it magically creating a new error for an attribute that doesn't exist, see https://github.com/rails/rails/blob/master/activemodel/lib/active_model/errors.rb#L204 for the behavior. This seems to be expected but I may be wrong.

**Comment by lukesarnacki** [comment]
@h0jeZvgoxFepBQ2C I was looking into this and it is fixed in rails 6 (`@details` and `@messages` were removed and the implementation of the lookup was changed). Not sure what core team members would say but I guess it is pointless to patch something that is already fixed in next versions.

**Comment by rafaelfranca** [comment]
Closing since it is fixed on Rails 6 and Rails 5.2 is not accepting bug fixes anymore. Thank you for the issue
